Update margin check to only apply for opening positions or decreasing collateral

prateekdefi opened the pull request:
Issue: https://linear.app/perennial/issue/PE-2296/[fix]-cant-close-position-when-between-margin-and-maintenance

Updated margin check to only apply if opening a position or decreasing collateral. (Updating margin check for only opening position was resulting in the user withdrawing all collateral even with `position != 0`)

EdNoepel reviewed (commented):
Few questions:
- Regarding the collateral check, is there a reason we still want to check margin if _not_ changing position and increasing collateral?
- Could you provide a test exercising the situation where user is between maintenance and margin requirements, showing this change allows user to reduce/close their position?

prateekdefi commented:
> *  Regarding the collateral check, is there a reason we still want to check margin if _not_ changing position and _increasing_ collateral?

I added this check because when a user adds collateral, it's important to ensure that the position meets the required collateral margin. Even though the position itself is not changing, by increasing collateral, we treat the whole situation as a new order and verify that it satisfies the margin requirement. 

> * Could you provide a test exercising the situation where user is between maintenance and margin requirements, showing this change allows user to reduce/close their position?

Added a test [0ffcff1](https://github.com/equilibria-xyz/perennial-v2/pull/591/commits/0ffcff1763f5a1abd7696cbbedb95fbbe8fc0727), where a user can close its position below margin requirements

kbrizzle commented:
> I added this check because when a user adds collateral, it's important to ensure that the position meets the required collateral margin. Even though the position itself is not changing, by increasing collateral, we treat the whole situation as a new order and verify that it satisfies the margin requirement.

@prateekdefi This is kind of the trap we landed in originally creating this margin-maintenance issue. Even though technically the new resulting position may be under the margin requirement, we want to always allow the user to do actions which "improve" their position. So in this case we should all deposits even if the deposit is not quite enough to make it to margin (it may be enough to stop a liquidation from happening).

EdNoepel reviewed (dismissed):
LGTM

prateekdefi commented on packages/core/test/integration/core/happyPath.test.ts:
```diff
@@ -2934,6 +2934,42 @@ describe('Happy Path', () => {
     })
   })
 
+  it('user can close position even if collateral less than margin required', async () => {
```
Added in [d1018c6](https://github.com/equilibria-xyz/perennial-v2/pull/591/commits/d1018c664ff90977a0bc412817c8a4d79d85bb56)
